### PR TITLE
added implementation and integration test

### PR DESCRIPTION
**Pull request by steven-xu-lf**
- added initial implementation of a method called `static async getSelfHostedRepositoryList`
- added a conditional integration tests that will only run if it is running in a self hosted environment (tested locally)
- updated changelog and version

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -12651,17 +12651,32 @@ export interface ILinkDefinitionsClient {
...
 
-export class ApiException extends ApiExceptionCore  {
-  constructor(message: string, status: number, response: string, headers: { [key: string]: any; }, result: any) {
+export class RepositoryClient extends RepositoriesClient {
+  /**
+   * Returns the repository resource list that current user has access to given the API server base URL. Only available in Laserfiche Self-Hosted.
+   * @param args.baseUrl API server base URL e.g., https://{APIServerName}/LFRepositoryAPI
+   * @returns Get the repository resource list successfully.
+   */
+  public static async getSelfHostedRepositoryList(args: { baseUrl: string }): Promise<RepositoryInfo[]> {
```
Comment (jason-chan-lf) [inline src/index.ts:12660]: it looks like this function got added into a new class `RepositoryClient` instead of being in the `RepositoriesClient`

Comment (steven-xu-lf) [reply src/index.ts:12660]: yea, the auto generated `index.ts` file doesn't allow duplicate class names 

image

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 1.1.4 Pre-release
+
+### Fixes
```
[the remark this answers is not in the document]
Comment (steven-xu-lf) [reply CHANGELOG.md]: ok done

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 1.1.4 Pre-release
+
+### Features
+- Added `RepositoryClient.getSelfHostedRepositoryList` method that will enable self hosted users to get their repository list without an access token.
```
Comment (jason-chan-lf) [inline CHANGELOG.md:4]: update to `RepositoriesClient`

**Pull request merged**